From 799d0e9399a671f9a99c402cc7ac7008fa76fa37 Mon Sep 17 00:00:00 2001
From: Derek Foreman <derek.foreman@collabora.com>
Date: Mon, 16 Sep 2024 13:09:00 -0500
Subject: [PATCH 187/187] [WIP-HDR]: gpu: hdmi: Always install colorspace
 connector property

The colorspace property let user read/set the supported colorspaces which
should be filtered against the EDID data. Without this, the compositors
cannot know and signal its supported and what is going to be used.

FIXME: We also trigger an update on HDR data change, should be its own
patch

Signed-off-by: Nicolas Dufresne <nicolas.dufresne@collabora.com>
---
 drivers/gpu/drm/display/drm_hdmi_state_helper.c | 4 +++-
 drivers/gpu/drm/drm_connector.c                 | 6 ++++++
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/display/drm_hdmi_state_helper.c b/drivers/gpu/drm/display/drm_hdmi_state_helper.c
index 029919354da7..4288ba6cd829 100644
--- a/drivers/gpu/drm/display/drm_hdmi_state_helper.c
+++ b/drivers/gpu/drm/display/drm_hdmi_state_helper.c
@@ -876,7 +876,9 @@ int drm_atomic_helper_connector_hdmi_check(struct drm_connector *connector,
 
 	if (old_conn_state->hdmi.broadcast_rgb != new_conn_state->hdmi.broadcast_rgb ||
 	    old_conn_state->hdmi.output_bpc != new_conn_state->hdmi.output_bpc ||
-	    old_conn_state->hdmi.output_format != new_conn_state->hdmi.output_format) {
+	    old_conn_state->hdmi.output_format != new_conn_state->hdmi.output_format ||
+	    old_conn_state->hdr_output_metadata != new_conn_state->hdr_output_metadata ||
+	    old_conn_state->colorspace != new_conn_state->colorspace) {
 		struct drm_crtc *crtc = new_conn_state->crtc;
 		struct drm_crtc_state *crtc_state;
 
diff --git a/drivers/gpu/drm/drm_connector.c b/drivers/gpu/drm/drm_connector.c
index 1bec7e5c2215..ca0a43196abd 100644
--- a/drivers/gpu/drm/drm_connector.c
+++ b/drivers/gpu/drm/drm_connector.c
@@ -640,6 +640,12 @@ int drmm_connector_hdmi_init(struct drm_device *dev,
 	if (!drm_mode_create_color_format_property(connector, supported_drm_formats))
 		drm_connector_attach_color_format_property(connector);
 
+        ret = drm_mode_create_hdmi_colorspace_property(connector, 0);
+        if (ret)
+                return ret;
+
+        drm_connector_attach_colorspace_property(connector);
+
 	connector->hdmi.funcs = hdmi_funcs;
 
 	return 0;
-- 
2.43.0

